tf_encrypted/keras/backend.py

```diff
@@ -0,0 +1,62 @@
+"""TFE Keras backend. Most of the code was
+borrowed from the tf.keras codebase.
+"""
+import threading
+
+from tensorflow.python.framework import ops
+from tensorflow.python.keras.backend import get_graph, name_scope
+from tensorflow.python.ops import array_ops
+
+import tf_encrypted as tfe
+
+# This is a thread local object that will hold the default internal TFE session
+# used by TFE Keras. It can be set manually via `set_session(sess)`.
+_SESSION = threading.local()
+
+
+def get_session(op_input_list=()):
+  """Returns the session object for the current thread."""
+  global _SESSION
+  default_session = ops.get_default_session()
+  if default_session is not None:
+    session = default_session
+    # If the default session is not a TFE Session, create one
+    if not isinstance(session, tfe.Session()):
```

Comment:
i think there's still a flaw in the logic here. if `default_session` is a TF session, then any TFE session already in `_SESSION` will be ignored and a new one created.

maybe something like:
1. if `_SESSION.session` is not none then return that
2. if `ops.get_default_session` is a TFE session then return that
3. create a new TFE session, store in `_SESSION.session`, and return that

Reply:
my bad, apologize, tried to fix this too quickly yesterday. good catch. thanks